strawberry/annotation.py

```diff
@@ -101,19 +101,17 @@ def annotation(self) -> Union[object, str]:
     def annotation(self, value: Union[object, str]) -> None:
         self.raw_annotation = value
 
+        self.__resolved_cached__ = None
+
     def evaluate(self) -> type:
```

Comment:
this is used in a couple of places directly, so we might want to reintroduce the cache, but I need to double check why we use it directly

Reply:
I think we only use it here: https://github.com/strawberry-graphql/strawberry/blob/095463d4d182bfb94c26b08f333b9be31eab359a/strawberry/types/fields/resolver.py#L121

@skilkis do you think it is worth caching the result here too? 😊